src/test/java/org/dmfs/jems/single/elementary/DigestedTest.java

```diff
@@ -0,0 +1,100 @@
+        assertThat(new Digested(new Md5(), "latin1", (CharSequence) "1äöü", "456").value(), is(new Md5().newInstance().digest("1äöü456".getBytes("latin1"))));
+        assertThat(new Digested(new Md5(), "latin1", (CharSequence) "1äöü", "456", "789").value(),
+                is(new Md5().newInstance().digest("1äöü456789".getBytes("latin1"))));
+
+        assertThat(new Digested(new Md5(), new ValueSingle<>(new byte[0])).value(), is(new Md5().newInstance().digest()));
+        assertThat(new Digested(new Md5(), new ValueSingle<>(new byte[0]), new ValueSingle<>(new byte[0])).value(), is(new Md5().newInstance().digest()));
+        assertThat(new Digested(new Md5(), new ValueSingle<>(new byte[0]), new ValueSingle<>(new byte[0]), new ValueSingle<>(new byte[0])).value(),
+                is(new Md5().newInstance().digest()));
+
+        assertThat(new Digested(new Md5(), new ValueSingle<>(new byte[] { 1, 2, 3 })).value(), is(new Md5().newInstance().digest(new byte[] { 1, 2, 3 })));
+        assertThat(new Digested(new Md5(), new ValueSingle<>(new byte[] { 1, 2, 3 }), new ValueSingle<>(new byte[] { 4, 5, 6 })).value(),
+                is(new Md5().newInstance().digest(new byte[] { 1, 2, 3, 4, 5, 6 })));
+        assertThat(new Digested(new Md5(), new ValueSingle<>(new byte[] { 1, 2, 3 }), new ValueSingle<>(new byte[] { 4, 5, 6 }),
+                new ValueSingle<>(new byte[] { 7, 8, 9 })).value(), is(new Md5().newInstance().digest(new byte[] { 1, 2, 3, 4, 5, 6, 7, 8, 9 })));
+
+        assertThat(new Digested(new Md5(), new Seq<Single<byte[]>>(new ValueSingle<>(new byte[0]))).value(), is(new Md5().newInstance().digest()));
+        assertThat(new Digested(new Md5(), new Seq<Single<byte[]>>(new ValueSingle<>(new byte[0]), new ValueSingle<>(new byte[0]))).value(),
+                is(new Md5().newInstance().digest()));
+        assertThat(
+                new Digested(new Md5(), new Seq<Single<byte[]>>(new ValueSingle<>(new byte[0]), new ValueSingle<>(new byte[0]), new ValueSingle<>(new byte[0])))
+                        .value(), is(new Md5().newInstance().digest()));
+
+        assertThat(new Digested(new Md5(), new Seq<Single<byte[]>>(new ValueSingle<>(new byte[] { 1, 2, 3 }))).value(),
+                is(new Md5().newInstance().digest(new byte[] { 1, 2, 3 })));
+        assertThat(
+                new Digested(new Md5(), new Seq<Single<byte[]>>(new ValueSingle<>(new byte[] { 1, 2, 3 }), new ValueSingle<>(new byte[] { 4, 5, 6 }))).value(),
+                is(new Md5().newInstance().digest(new byte[] { 1, 2, 3, 4, 5, 6 })));
+        assertThat(new Digested(new Md5(), new Seq<Single<byte[]>>(new ValueSingle<>(new byte[] { 1, 2, 3 }), new ValueSingle<>(new byte[] { 4, 5, 6 }),
+                new ValueSingle<>(new byte[] { 7, 8, 9 }))).value(), is(new Md5().newInstance().digest(new byte[] { 1, 2, 3, 4, 5, 6, 7, 8, 9 })));
+    }
+
+
+    @Test
+    public void testIllegalEncoding()
```

Comment:
Why not just use the annotation?
```java
    @Test(expected = RuntimeException.class)
    public void testIllegalEncoding()
    {
        new Digested(new Md5(), "bogus-fake-invalid", (CharSequence) "").value();
    }
```

Reply:
I want to make sure it's the `value()` method which throws, not something in the constructor.

Reply:
oh, I see